pavelib/quality.py

```diff
@@ -230,103 +125,45 @@ def run_quality(options):
     :param: p, diff-quality will fail if the quality percentage calculated is
         below this percentage. For example, if p is set to 80, and diff-quality finds
         quality of the branch vs the compare branch is less than 80%, then this task will fail.
-        This threshold would be applied to both pep8 and pylint.
+        This threshold would be applied to both PEP8 and Pylint.
     """
 
-    # Directory to put the diff reports in.
-    # This makes the folder if it doesn't already exist.
-    dquality_dir = (Env.REPORT_DIR / "diff_quality").makedirs_p()
-    diff_quality_percentage_failure = False
+    directory_report = get_clean_report_directory('diff_quality')
+    percentage_failure = False
 
-    # Set the string, if needed, to be used for the diff-quality --compare-branch switch.
-    compare_branch = getattr(options, 'compare_branch', None)
-    compare_branch_string = ''
+    compare_branch = getattr(options, 'compare_branch', '')
     if compare_branch:
-        compare_branch_string = '--compare-branch={0}'.format(compare_branch)
+        compare_branch = "--compare-branch={0}".format(compare_branch)
 
-    # Set the string, if needed, to be used for the diff-quality --fail-under switch.
     diff_threshold = int(getattr(options, 'percentage', -1))
     percentage_string = ''
     if diff_threshold > -1:
         percentage_string = '--fail-under={0}'.format(diff_threshold)
 
-    # Generate diff-quality html report for pep8, and print to console
-    # If pep8 reports exist, use those
-    # Otherwise, `diff-quality` will call pep8 itself
-
-    pep8_files = get_violations_reports("pep8")
-    pep8_reports = u' '.join(pep8_files)
-
-    try:
-        sh(
-            "diff-quality --violations=pep8 {pep8_reports} {percentage_string} "
-            "{compare_branch_string} --html-report {dquality_dir}/diff_quality_pep8.html".format(
-                pep8_reports=pep8_reports,
-                percentage_string=percentage_string,
-                compare_branch_string=compare_branch_string,
-                dquality_dir=dquality_dir
-            )
-        )
-    except BuildFailure, error_message:
-        if is_percentage_failure(error_message):
-            diff_quality_percentage_failure = True
-        else:
-            raise BuildFailure(error_message)
-
-    # Generate diff-quality html report for pylint, and print to console
-    # If pylint reports exist, use those
-    # Otherwise, `diff-quality` will call pylint itself
+    pep8_count = pep8()
+    percentage_failure = pep8_count > 0
```

Comment:
@stvstnfrd I want the diff-quality job to still provide a report that people can see in HTML form when they view their results on Jenkins. Ideally what I want is all the violations printed out together that you can see when you view the diff-quality report.

I've been working on this at #7352 . Can you revert this change, and make this PR **only** about Pylint? I think it'd be better to separate the ideas anyway. If you want to pick up finishing my work from #7352 or fix pep8 in a separate PR, feel free to do that, but let's keep this pr focused.